### ISSUE TITLE
[DEPRECATION] missing data for `sync` relationships should not be considered `empty`

### ISSUE THREAD

**Issue by runspired** [emberjs/data#8816]
Currently, if a synchronous relationship receives a payload containing no information about it's state, we initialize the relationship as if it were canonically empty. This is incorrect, as we don't actually know the state (having received no information about it), which leads to us being unable to utilize `reference.load()` which would otherwise allow a user to utilize `findBelongsTo`/`findHasMany` to load the relationship manually. While users of references may use `reference.reload()` instead, we should also firm up our API expectations here, as this is a place wherein we deviate from json-api.

This deprecation would warn when this scenario is encountered. Once the deprecation lifecycle is complete, this deprecation would turn into an assertion, as the relationship is in an invalid state. For additional context, see this bugfix commit that restored this behavior after it was unintentionally broken/discovered: https://github.com/emberjs/data/pull/5533/commits/850bdf2688fe9e9cbfb4f381cb9d460acf7150f7

**Comment by runspired** [comment]
closing in favor of https://github.com/emberjs/data/issues/7584
